fahrschein-http-jdk11/src/test/java/org/zalando/fahrschein/http/jdk11/JavaNetHeadersDelegateTest.java

```diff
@@ -0,0 +1,101 @@
+package org.zalando.fahrschein.http.jdk11;
+
+import org.junit.Test;
+import org.zalando.fahrschein.http.api.ContentType;
+import org.zalando.fahrschein.http.api.Headers;
+
+import java.net.URI;
+import java.net.http.HttpRequest;
+import java.util.HashSet;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptySet;
+import static java.util.Collections.singletonList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class JavaNetHeadersDelegateTest {
+
+    private HttpRequest.Builder requestBuilder() {
+        return HttpRequest.newBuilder(URI.create("http://bla"));
+    }
+
+    @Test
+    public void shouldBeCaseInsensitive() {
+        final Headers headers = new JavaNetHeadersDelegate(HttpRequest.newBuilder()
+                .uri(URI.create("http://bla"))
+                .setHeader("Content-type", "text/plain").build().headers());
+
+        assertEquals(singletonList("text/plain"), headers.get("content-type"));
+        assertEquals("text/plain", headers.getFirst("CONTENT-TYPE"));
+        assertEquals("text/plain", headers.getFirst("Content-Type"));
+        assertEquals("text/plain", headers.getFirst("content-type"));
+    }
+
+    @Test
+    public void shouldReturnHeaderNames() {
+        final Headers headers = new JavaNetHeadersDelegate(requestBuilder()
+                .setHeader("Content-Type", "text/plain")
+                .setHeader("Content-Encoding", "gzip").build().headers());
+
+        assertEquals(new HashSet<>(asList("Content-Type", "Content-Encoding")), headers.headerNames());
+    }
+
+    @Test
+    public void shouldReturnEmptySetOfHeadernames() {
+        final Headers headers = new JavaNetHeadersDelegate(requestBuilder().build().headers());
+        assertEquals(emptySet(), headers.headerNames());
+    }
+
+    @Test
+    public void shouldReturnEmptyListForUnknownHeaders() {
+        final Headers headers = new JavaNetHeadersDelegate(requestBuilder().build().headers());
+        assertEquals(emptyList(), headers.get("Content-Type"));
```

Comment:
Let's use the constants defined in `org.zalando.fahrschein.http.api.Headers`.
I also suggest adding `Content-Encoding` there, and replacing the occurrences (also fine to delegate the replacement in other modules to a different PR, but it's quite a simple task I'd do it here).

The constants can also be used wherever you're testing with upper or lower case.

Reply:
I've started doing this in the other PR: https://github.com/zalando-nakadi/fahrschein/pull/320 - I'll rebase this one once the zstd PR is merged.